### DISCUSSION TITLE
Always occur errors say `ERROR [ExceptionHandler] no pg_hba.conf entry for host`

### DISCUSSION THREAD

**Discussion by kaiwenxiao** [mikro-orm/mikro-orm#5953] [Q&A] [answered]
### Describe the bug

Hi, I'm using Nestjs + MikroOrm + Postgresql. And I try the [aiven.io](aiven.io) as my remote database provider.

When I start the project always blame me about `[Nest] 6096  - 08/18/2024, 3:32:06 PM   ERROR [ExceptionHandler] no pg_hba.conf entry for host "218.18.130.180", user "avnadmin", database "defaultdb", no encryption`

May you help to check out my issue? Thanks!

### Reproduction

- This my Github Repository: https://github.com/kaiwenxiao/utter-nest

- files may need to introspect:

1. `src/common/database/orm.config.ts`
2. `src/common/database/mikro-orm-cli.config.ts`
3. `src/lib/config/configs/database.config.ts`

- In my computer, I install Postgresql 16 and my `pg_hba.conf` config sort of likes this:

![image](https://github.com/user-attachments/assets/e09e204d-8941-43af-af34-1c5369f62c07)

Any more info need to provide just info it, THANKS!

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6

### Node.js version

20

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by MilanKovacic** [comment]
In the configuration, try to set the following option:

```
  driverOptions: {
    connection: { ssl: true },
  },
```

**Comment by kaiwenxiao** [comment]
@MilanKovacic Thanks a lot. provided by your hint, I'm using
`driverOptions: {
   {
     connection: { ssl: { rejectUnauthorized: false } }
   }
},`
from https://github.com/mikro-orm/mikro-orm/issues/303#issuecomment-1100969462.

---

BTW, I get another question about `[Nest] 11356  - 08/19/2024, 9:14:43 PM   ERROR [ExceptionHandler] Metadata for entity ./dist/entities/**/*.entity.js not found`, and I using the recommond `entities` and `entitiesTs` config from the official document:
![image](https://github.com/user-attachments/assets/d7b1938c-0e25-4409-bcc1-8c6ada796cfe)

And also `package.json` config:
`"mikro-orm": {
    "useTsNode": true,
    "configPaths": [
      "./src/common/database/mikro-orm-cli.config.ts",
      "./dist/common/database/mikro-orm-cli.config.js"
    ]
  },`

Could you help with this? Thanks

**Reply by B4nan** [reply]
Hard to help with a screenshot, can you at least provide a complete stack trace? 

You could take a look at https://github.com/mikro-orm/nestjs-realworld-example-app which should have everything set up correctly. 

> And also package.json config:
"mikro-orm": { "useTsNode": true, "configPaths": [ "./src/common/database/mikro-orm-cli.config.ts", "./dist/common/database/mikro-orm-cli.config.js" ] },

This is only for the CLI.

**Reply by kaiwenxiao** [reply]
@B4nan  Thanks for quick reply. I changed the config as the demo you show, but still not working.
![image](https://github.com/user-attachments/assets/f652d7a0-fd62-49a7-a389-b3b7921b4abc)

src/common/database/mikro-orm-cli.config.ts:

```
const config: Options = defineConfig({
  ...baseOptions,
  clientUrl: process.env.DB_CLIENTURL,
  // dbName: process.env.DB_DATABASE,
  // password: process.env.DB_PASSWORD,
  // port: process.env.DB_PORT as number | undefined,
  // user: process.env.DB_USERNAME,
  // host: process.env.DB_HOST,
  extensions: [Migrator, SeedManager],
});
```

**Reply by B4nan** [reply]
That error is not coming from the ORM init, but from nest DI container trying to create a repository for entity called `./dist/entities/**/*.entity.js` which is obviously wrong. Sounds like you are doing something like `@InjectRepository('./dist/entities/**/*.entity.js')`. Also I guess the stack trace is not complete either, I would really appreciate if you copy the complete logs instead of screenshoting.

The example app I shared won't produce this.

**Reply by kaiwenxiao** [reply]
Definitely sure. This is the whole error trace

```
[Nest] 14008  - 08/19/2024, 10:23:16 PM   ERROR [ExceptionHandler] Metadata for entity dist/**/*.entity.js not found
MetadataError: Metadata for entity dist/**/*.entity.js not found
    at MetadataError.missingMetadata (C:\software\project\utter-nest\node_modules\.pnpm\@mikro-orm+core@6.3.3\node_modules\@mikro-orm\core\errors.js:188:16)
    at MetadataStorage.get (C:\software\project\utter-nest\node_modules\.pnpm\@mikro-orm+core@6.3.3\node_modules\@mikro-orm\core\metadata\MetadataStorage.js:54:42)
    at SqlEntityManager.getRepository (C:\software\project\utter-nest\node_modules\.pnpm\@mikro-orm+core@6.3.3\node_modules\@mikro-orm\core\EntityManager.js:89:40)
    at SqlEntityManager.getRepository (C:\software\project\utter-nest\node_modules\.pnpm\@mikro-orm+knex@6.3.3_@mikro-orm+core@6.3.3_pg@8.12.0\node_modules\@mikro-orm\knex\SqlEntityManager.js:32:22)
    at InstanceWrapper.useFactory [as metatype] (C:\software\project\utter-nest\node_modules\.pnpm\@mikro-orm+ne
stjs@6.0.2_@mikro-orm+core@6.3.3_@nestjs+common@10.3.10_@nestjs+core@10.3.10\node_modules\@mikro-orm\nestjs\mikro-orm.providers.js:106:34)
    at Injector.instantiateClass (C:\software\project\utter-nest\node_modules\.pnpm\@nestjs+core@10.3.10_@nestjs
+common@10.3.10_@nestjs+platform-express@10.3.10_@nestjs+websocke_4gxtsgmlgbsjgkaqaywtwkdvwi\node_modules\@nestjs\core\injector\injector.js:368:55)
    at callback (C:\software\project\utter-nest\node_modules\.pnpm\@nestjs+core@10.3.10_@nestjs+common@10.3.10_@
nestjs+platform-express@10.3.10_@nestjs+websocke_4gxtsgmlgbsjgkaqaywtwkdvwi\node_modules\@nestjs\core\injector\injector.js:65:45)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Injector.resolveConstructorParams (C:\software\project\utter-nest\node_modules\.pnpm\@nestjs+core@1
0.3.10_@nestjs+common@10.3.10_@nestjs+platform-express@10.3.10_@nestjs+websocke_4gxtsgmlgbsjgkaqaywtwkdvwi\node_modules\@nestjs\core\injector\injector.js:144:24)
    at async Injector.loadInstance (C:\software\project\utter-nest\node_modules\.pnpm\@nestjs+core@10.3.10_@nest
js+common@10.3.10_@nestjs+platform-express@10.3.10_@nestjs+websocke_4gxtsgmlgbsjgkaqaywtwkdvwi\node_modules\@nestjs\core\injector\injector.js:70:13)
```

**Reply by kaiwenxiao** [reply, marked as answer]
@B4nan I fix the problem by myself, just remove the redundant
```
MikroOrmModule.forFeature({
  entities: baseOptions.entities,
}),
```
in `MikroOrmModule` setup:

```
@Global()
@Module({
  imports: [
    MikroOrmModule.forRootAsync({
      imports: [ConfigModule],
      inject: [ConfigService],
      useFactory: (configService: ConfigService<Configs, true>) => {
        return defineConfig({
          ...baseOptions,
          clientUrl: configService.get('database.clientUrl', { infer: true }),
          // host: configService.get('database.host', { infer: true }),
          // port: configService.get('database.port', { infer: true }),
          // dbName: configService.get('database.dbName', { infer: true }),
          // user: configService.get('database.user', { infer: true }),
          // password: configService.get('database.password', { infer: true }),
        })
      }
    }),
    // MikroOrmModule.forFeature({
    //   entities: baseOptions.entities,
    // }),
  ],
  exports: [MikroOrmModule],
})
export class OrmModule {}
```

**Reply by B4nan** [reply]
Its not just redundant, its wrong, you are supposed to use class reference or entity name in that place, not a glob.
